### PR TITLE
fix: pagination componnet is not displayed at the position is none of…

### LINKED ISSUE #29235: Table pagination overrides setting top and bottom position to 'none'
- [X] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### Reproduction link
[https://ant.design/components/table/#components-table-demo-dynamic-settings](https://ant.design/components/table/#components-table-demo-dynamic-settings)

### Steps to reproduce
Pagination config option for the Table component allows setting position for top and bottom pagination component like so:

```<Table pagination={{ position: ['topLeft', 'bottomLeft']}} />```

This position array is painfully confusing. It can contain up to two strings prefixed with `top` or `bottom`, placing up to two pagination components for the top and bottom according to the first two valid positions it sees.

In the documentation, it offers the choice to turn off pagination completely by replacing the config object with `false`. But it also shows the option of setting these positions to `'none'`, which works for one of the positions but is disregarded when set to _both_.

### What is expected?
```<Table pagination={{ position: ['none', 'none']}} />```

Should disable pagination component.

### What is actually happening?
Pagination component is displayed at the default position of bottom right of the table.

| Environment | Info |
|---|---|
| antd | 4.12.2 |
| React | 17.0.1 |
| System | macOS 10.15.7 |
| Browser | Chrome 88.0 / Firefox 85.0 |

### PR DESCRIPTION
**Pull request by mumiao**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [x] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link
close https://github.com/ant-design/ant-design/issues/29235

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |   fix Pagination componnet is not displayed at the position is none of array   |
| 🇨🇳 Chinese |  修复 Table Pagination 组件定位都设置为none时，Pagination组件不显示     |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

**Comment by mumiao** [comment]
`position: ['none', 'none']` 或者其他不符合 `topLeft topRight topCenter bottomLeft bottomCenter bottomRight`
统一翻译为：`position: ['invalid', 'invalid'] `

File: components/table/__tests__/Table.pagination.test.js [hunk 1]
```diff
@@ -325,12 +325,14 @@ describe('Table.pagination', () => {
     expect(wrapper.find('.ant-spin-container').childAt(0).find('.ant-pagination')).toHaveLength(1);
     expect(wrapper.find('.ant-spin-container').childAt(2).find('.ant-pagination')).toHaveLength(1);
     wrapper.setProps({ pagination: { position: ['invalid'] } });
-    expect(wrapper.find('.ant-pagination')).toHaveLength(1);
+    expect(wrapper.find('.ant-pagination')).toHaveLength(0);
+    wrapper.setProps({ pagination: { position: ['invalid', 'invalid'] } });
+    expect(wrapper.find('.ant-pagination')).toHaveLength(0);
```
Comment (afc163) [inline components/table/__tests__/Table.pagination.test.js:330]: 你的测试用例应该是两个 none 的时候，都不显示。

一个或者两个 invalid 是应该显示默认的 .ant-pagination 的。

Comment (mumiao) [reply components/table/__tests__/Table.pagination.test.js:330]: 之前设计不符合 topLeft bottomLeft 等 position统一默认显示右边，而invalid none都不符合 
我理解 none invalid 或者其他任何不符合的都要禁用Pagination，代码默认的逻辑我直接干掉了
按照您这边提的case 我要单独对position=invalid进行逻辑判断？然后其他非invalid或者非topLeft等都禁用？

Comment (afc163) [reply components/table/__tests__/Table.pagination.test.js:330]: 应该是非法的配置都视为无效，而不是视为 none。

Comment (mumiao) [reply components/table/__tests__/Table.pagination.test.js:330]: image
那我觉得我这个test case没有问题呢  非法的都视为无效 禁用了pagination

Comment (mumiao) [reply components/table/__tests__/Table.pagination.test.js]: done

**Comment by mumiao** [comment]
resolved

**Pull request merged**